### PR TITLE
Remove code for reading notebooks as XML

### PR DESCRIPTION
**Pull request by takluyver**
Leave an explanatory error message on the off chance that anyone is still using it.

Closes gh-132

### CODE REVIEW

File: nbformat/v2/nbxml.py [hunk 1]
```diff
@@ -1,189 +1,26 @@
...
-
-def _get_int(e, tag):
-    sub_e = e.find(tag)
-    if sub_e is None:
-        return None
-    else:
-        return int(sub_e.text)
 
+REMOVED_MSG = """\
+Reading notebooks as XML has been removed as security issues were discovered.
```
Comment (willingc) [inline nbformat/v2/nbxml.py]: ... removed to harden security and possible DoS with XML use.

File: nbformat/v2/nbxml.py [hunk 2]
```diff
@@ -1,189 +1,26 @@
...
 
-def _set_int(nbnode, attr, parent, tag):
-    if attr in nbnode:
-        e = ET.SubElement(parent, tag)
-        e.text = unicode_type(nbnode[attr])
+The XML notebook format was deprecated before the Jupyter (previously IPython)
+Notebook was ever released. We are not aware of anyone using it, so we have
+removed it rather than trying to fix the problems.
```
Comment (willingc) [inline nbformat/v2/nbxml.py]: ...it. ~~rather than trying to fix the problems.~~

File: nbformat/v2/nbxml.py [hunk 3]
```diff
@@ -1,189 +1,26 @@
...
 
+If you were using this code, and you need to continue using it, feel free to
+fork an earlier version of the nbformat package and maintain it yourself.
```
Comment (willingc) [inline nbformat/v2/nbxml.py]: :+1:

File: nbformat/v2/nbxml.py [hunk 4]
```diff
@@ -1,189 +1,26 @@
...
+The issue which prompted its removal is:
```
Comment (willingc) [inline nbformat/v2/nbxml.py]: its -> this

**Comment by takluyver** [comment]
This PR also now drops Python 3.3 from the tests on Travis - there was an error there, and seeing as [3.3 is EOL now](https://devguide.python.org/#status-of-python-branches), it doesn't seem worth solving that.

In practical terms, the code will probably go on working on 3.3 for the foreseeable future; nbformat is pretty plain code, and it's still tested on both 3.4 and 2.7, so we're unlikely to do anything that breaks 3.3. But we're no longer testing it.

**Comment by takluyver** [comment]
I'll merge this; several responses so far have been unanimously in favour.

**Pull request merged**